src/arrow_field.rs

```diff
@@ -0,0 +1,209 @@
+use crate::reader::get_schema;
+use capnp::dynamic_value;
+use capnp::introspect::{RawEnumSchema, TypeVariant};
+use capnp::schema::{Field as CapnpField, StructSchema};
+use capnp::Error as CapnpError;
+use polars_arrow::datatypes::{ArrowDataType, Field as ArrowField, IntegerType};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+
+type Result<T> = std::result::Result<T, Error>;
+
+enum Error {
+    Capnp(CapnpError),
+    EmptyStruct,
+    RecursionLimitExceeded,
+}
+
+pub static ENUMERANTS_METADATA_KEY: &str = "enumerants";
```

Comment:
Could these be `const` instead of `static`? iirc that will cause them to be inlined where used rather than referenced, which is something to consider, but I don't think it would be a problem here.